src/com/thoughtworks/gauge/execution/GaugeCommandLine.java

```
@@ -0,0 +1,30 @@
+package com.thoughtworks.gauge.execution;
+
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.Project;
+import com.thoughtworks.gauge.Constants;
+import com.thoughtworks.gauge.exception.GaugeNotFoundException;
+import com.thoughtworks.gauge.settings.GaugeSettingsModel;
+import com.thoughtworks.gauge.util.GaugeUtil;
+
+import java.util.Map;
+
+public class GaugeCommandLine {
+    public static GeneralCommandLine getInstance(Module module, Project project) {
+        GeneralCommandLine commandLine = new GeneralCommandLine();
+        try {
+            GaugeSettingsModel settings = GaugeUtil.getGaugeSettings();
+            commandLine.setExePath(settings.getGaugePath());
+            Map<String, String> environment = commandLine.getEnvironment();
+            environment.put(Constants.GAUGE_HOME, settings.getHomePath());
+        } catch (GaugeNotFoundException e) {
+            commandLine.setExePath(Constants.GAUGE);
```

This assumes that gauge exe will be found in project/module folder. The exe will certainly not be found here. We should throw an exception in this case instead of setting defaults.

---

This is not assuming that `gauge` binary will be present in project/module folder. Instead, It is running the way a user would run in the command line i.e `gauge specs/`. If it finds the binary in `PATH` then it would run `<path>/gauge specs`.

---

`GaugeUtil.getGaugeSettings()` is used in many places to execute gauge commands. If gauge path is not set in Intellij and not found PATH, these commands fail. Gauge module is itself not built. I feel that this case will not occur, or will fail later on.